include/libopencm3/stm32/fdcan.h

```diff
@@ -0,0 +1,1420 @@
+/** @defgroup fdcan_defines FDCAN defines
+@ingroup STM32H7_defines
+@brief <b>libopencm3 Defined Constants and Types for STM32 FDCAN </b>
+@version 1.0.0
+LGPL License Terms @ref lgpl_license
+*/
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef LIBOPENCM3_STM32_FDCAN_H
+#define LIBOPENCM3_STM32_FDCAN_H
+
+#include <libopencm3/cm3/common.h>
+#include <libopencm3/stm32/memorymap.h>
+
+/**@{
+ * @defgroup fdcan_reg_base CAN register base address
+ * @ingroup fdcan_defines
```

Comment:
you don't need any of these @ingroup fdcan_defines when it's the logical nesting level (like your line 1 group definition) you only need an @ingroup if you want to put it "elsewhere"

Reply:
I probably should just do a doxygen and see what happens; but since I don't have a `@file` in the first doxyblock in the file, nor do I have a `@{` are we still logically nesting at this point?

Reply:
yes, please do run doxygen yourself and see that it looks how you expect :)  Probably easiest to open a group around the file, we do that a lot elsewhere.